### ISSUE TITLE
Trying to launch editor fails

### ISSUE THREAD

**Issue by asmeurer** [ipython/ipython#9604]
```
In [2]: def test
  File "<ipython-input-2-50eadd3e41eb>", line 1
    def test
            ^
SyntaxError: invalid syntax

Return to editor to correct syntax error? [Y/n]
/bin/sh: -c: line 0: syntax error near unexpected token `newline'
/bin/sh: -c: line 0: `emacsclient -a 'emacs-server-start' -nw --socket-name=ttys000 +1 <ipython-input-2-50eadd3e41eb>'
/Users/aaronmeurer/anaconda/envs/ipython5/lib/python3.5/site-packages/IPython/terminal/ptshell.py:472: UserWarning: Could not open editor
  warn('Could not open editor')
```

My `$EDITOR` is `emacsclient -a 'emacs-server-start' -nw --socket-name=ttys000`. The command it is trying to run looks wrong, like it's trying to literally edit a file named `"<ipython-input-2-50eadd3e41eb>"`.

**Comment by asmeurer** [comment]
This is with `autoedit_syntax` enabled (see #9603).

**Comment by takluyver** [comment]
To clarify from #9603: this is not a regression, the bug was already there, but it was masked by the `autoedit_syntax` flag not working at all in previous versions.

It's supposed to trigger when there's a syntax error in a separate file, and there's code that tries to bypass it when the error comes from a prompt (see [here](https://github.com/ipython/ipython/blob/c963f6b4435fb150ebe995460649923b3ffc5e71/IPython/terminal/interactiveshell.py#L749)). But that code is out of date and doesn't pick up the filenames we now use for code typed at the prompt.

Playing devil's advocate a bit: `autoedit_syntax` has probably been broken for years, and we don't have lots of people complaining about it. We only discovered this because of something @asmeurer forgot in his config file. Maybe we should just ditch the feature.

**Comment by Carreau** [comment]
+1 to remove, let's get https://github.com/ipython/ipython/pull/9626 is as it will conflict.

**Comment by Carreau** [comment]
Closed by #9647 autoedit_syntax has now no effects. 
You can press up arrow and F2 to get into an editor. There is a current bug in PTK (fixed in master) that prevent `$EDITOR`with flags to work, but if you wrap in a shell script that takes no arguments it should work.

Thanks, feel free to reopen.
